### ISSUE TITLE
Closing Office dialog with ESC while Iframe element has focus leads to bug

### ISSUE THREAD

**Issue by michahell** [OfficeDev/office-js#9]
Exactly as title says: when you hit escape while focusing on an element in the dialog Iframe (say, an input or select) then dialog closes. The dialog can not be opened again.

## Expected Behavior

The dialog closes, ánd emits an event with error 12006 (dialog closed by user).

## Current Behavior

The dialog closes, and does not emit anything. on top of that, reopening the same dialog shows no dialog.
![screenshot 2018-12-05 at 15 16 37](https://user-images.githubusercontent.com/225784/49519454-4bcdc600-f8a1-11e8-9b4e-676c157af6ae.png)

**Comment by ATrogolo** [comment]
Has this issue been fixed? 
On Excel (Version: 1803 and Build: 9126.2295) if I open a dialog, press ESC and then trying to reopen the same dialog, it opens correctly.

**Comment by michahell** [comment]
Just checked. Yeap, absolutely. And completely ignored by MSFT.

![screenshot 2018-12-05 at 15 16 37](https://user-images.githubusercontent.com/225784/49519288-ef6aa680-f8a0-11e8-8255-cffa5ed4b0f0.png)

**Comment by michahell** [comment]
update: there is an **error** event being sent, code: **12007**.
But re-opening is still not possible... see screenshot above. So this is still a bug.

**Comment by kbrandl** [comment]
@michahell sorry to hear that you've having trouble with this, and apologies for the delayed response on this issue.

@JuaneloJuanelo are you able to provide any input on this issue?

**Comment by lliu113** [comment]
@michahell so sorry for getting back to you so late. I tried to repro this in Word (Version 2001, Build 12428.20000) and I was able to successfully reopen the dialog after closing with ESC. Are you still seeing this issue? 

Repro steps (I tried with the DocuSign add-in):
- Launch DocuSign
- Click "Log In". Dialog launches. 
- Click ESC. Dialog closes 
- Click "Log In". Dialog successfully reopens.

**Comment by DominikDoom** [comment]
This is absolutely still an issue, just today I encountered the same problem.
I can't verify if it happens in other clients too, but it happens exactly as OP described in Outlook (Windows 11, v. 2411, Build 18227.20162, MS 365).

The close event is never sent by the dialog if ESC is pressed, only if the dialog is closed with the window's X button or by explicitly calling .close() in the dialog API. After ESC the popup visually closes, but the reference seems to be retained somewhere in the background, effectively becoming a ghost instance. Stopping the addin webview process using Task manager or closing and reopening the addin task pane is the only thing that fixes it.

I tried working around this by listening to ESC keypress, unload and other similar events on the dialog web page and sending a message to the task pane parent requesting a manual close, but the messages were not received in this case. The window seems to be unloaded before that.
